packages/expo-av/src/Audio.ts

```diff
@@ -56,8 +58,8 @@ function getCurrentAudioMode(): AudioMode {
   return currentAudioMode;
 }
 
-export async function setAudioModeAsync(mode: AudioMode): Promise<void> {
-  mode = _populateMissingKeys(mode, getCurrentAudioMode());
+export async function setAudioModeAsync(partialMode: Partial<AudioMode>): Promise<void> {
+  const mode = _populateMissingKeys(partialMode, getCurrentAudioMode());
   currentAudioMode = mode;
```

Comment:
I know you didn't add this in, but looking at this now I'm thinking if we should refrain from setting `currentAudioMode` until after the checks that come directly after this 🤔

Reply:
Ah good catch! I just pushed a commit to move this down below after all the checks.